File: Src/Server/apcurium.MK.Web/admin/models/Rate.js
(function(){

    var Rate = TaxiHail.Rate = Backbone.Model.extend({
        urlRoot: TaxiHail.parameters.apiRoot + '/admin/rates',
        defaults: {
            name: '',
            flatRate: 0,
            pricePerPassenger: 0,
            distanceMultiplicator: 0,
            timeAdjustmentFactor: 20,
            startTime: null,
            endTime: null,
            daysOfTheWeek: 0
        }
    }, {
        type: {
            'default': 0,
            recurring: 1,
            day: 2
        }
    });

    Rate.prototype.defaults.type = Rate.type.recurring;

}());

File: Src/Server/MK.Common/Entity/Rate.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace apcurium.MK.Common.Entity
{
    public class Rate
    {
        public Guid Id { get; set; }
        public Guid CompanyId { get; set; }
        public string Name { get; set; }
        public decimal FlatRate { get; set; }
        public double DistanceMultiplicator { get; set; }
        public double TimeAdjustmentFactor { get; set; }
        public decimal PricePerPassenger { get; set; }
        public int DaysOfTheWeek { get; set; }
        public DateTime StartTime { get; set; }
        public DateTime EndTime { get; set; }
        public int Type { get; set; }
    }
}
